packages/remix-dev/cli/commands.ts

```diff
@@ -302,9 +302,11 @@ export async function dev(remixRoot: string, modeArg?: string) {
 }
 
 function purgeAppRequireCache(buildPath: string) {
-  for (let key in require.cache) {
-    if (key.startsWith(buildPath)) {
-      delete require.cache[key];
+  if (require.cache[buildPath + "/assets.json"]?.exports.version !== JSON.parse(require('fs').readFileSync(BUILD_DIR + '/assets.json')).version) {
```

Comment:
Could we make this an early return please?

This will create a positive condition (which has a lower cognitive load) + will make sure the code is less nested.